tidy3d/components/data/dataset.py

```diff
@@ -312,6 +312,88 @@ def apply_phase(self, phase: float) -> AbstractFieldDataset:
         return self
 
 
+class AuxFieldDataset(AbstractFieldDataset, ABC):
+    """Stores a collection of aux fields with x, y, z components."""
+
+    Nfx: Optional[EMScalarFieldType] = pd.Field(
+        None,
+        title="Nfx",
+        description="Spatial distribution of the free carrier density for "
+        "polarization in the x-direction.",
+    )
+    Nfy: Optional[EMScalarFieldType] = pd.Field(
+        None,
+        title="Nfy",
+        description="Spatial distribution of the free carrier density for "
+        "polarization in the y-direction.",
+    )
+    Nfz: Optional[EMScalarFieldType] = pd.Field(
+        None,
+        title="Nfz",
+        description="Spatial distribution of the free carrier density for "
+        "polarization in the z-direction.",
+    )
+
+    @property
+    def field_components(self) -> Dict[str, DataArray]:
+        """Maps the field components to their associated data."""
+        fields = {
+            "Nfx": self.Nfx,
+            "Nfy": self.Nfy,
+            "Nfz": self.Nfz,
+        }
+        return {field_name: field for field_name, field in fields.items() if field is not None}
+
+    @property
+    def grid_locations(self) -> Dict[str, str]:
+        """Maps field components to the string key of their grid locations on the yee lattice."""
+        return dict(Nfx="Ex", Nfy="Ey", Nfz="Ez")
+
+    @property
+    def symmetry_eigenvalues(self) -> Dict[str, Callable[[Axis], float]]:
+        """Maps field components to their (positive) symmetry eigenvalues."""
+
+        return dict(
+            Nfx=lambda dim: +1,
+            Nfy=lambda dim: +1,
+            Nfz=lambda dim: +1,
+        )
```

Comment:
I took another look at this PR to see if the symmetry expansion, and grid locations would be correctly handled and I'm pleasantly surprised that this seems to have been set up in a pretty expandable way previously! :D I was worried about it.